app/controllers/users_controller.rb

```diff
@@ -340,6 +362,16 @@ def prepare_privacy_settings
     @profile_visibility_users = User.find(current_user.setting(:profile_visibility, true).value(:users) || [])
   end
 
+  def prepare_newsletter_settings
+    @user = current_user
+    @newsletter_interval = current_user.newsletter_interval
+    @interval_options = [[I18n.t('users.settings.newsletter.interval.one'), '1'],
+                         [I18n.t('users.settings.newsletter.interval.week'), '7'],
+                         [I18n.t('users.settings.newsletter.interval.two_weeks'), '14'],
+                         [I18n.t('users.settings.newsletter.interval.month'), '30']
```

Comment:
When reading the source code, it wasn't very clear to me what an interval of "one" is. What about renaming it to "daily"? You could use these names for all intervals, e.g. daily, weekly, biweekly or fortnightly, monthly, etc. Just a suggestion...